monitor/monitor.go

```diff
@@ -13,7 +14,7 @@ import (
 
 // Watcher interface definition.
 type Watcher interface {
-	Watch(ctx context.Context, ch chan<- []*change.Change) error
+	Watch(ctx context.Context) (<-chan []change.Change, error)
```

Comment:
This is the goal of the change: 

The watcher should be the one to create and handle closing the channel since he is the data producer.